### ISSUE TITLE
False positive need-check-nil when using or assignment for default value

### ISSUE THREAD

**Issue by cyril0124** [EmmyLuaLs/emmylua-analyzer-rust#937]
<img width="632" height="181" alt="Image" src="https://github.com/user-attachments/assets/62cf86e0-6c6a-4229-86df-7593262a4d74" />

```lua
---@param v fun()?
local function test(v)
    v = v or function()
    end

    v() -- Should not report `need-check-nil`
end

test()
```

It works correctly in version 0.19.0, but this error occurs in the latest version (0.20.0).

**Comment by xuhuanzy** [comment]
The current version should be correct.
